tests/openwisp2/sample_notifications/tests/test_django.py

```diff
@@ -46,6 +53,30 @@ def test_test_app_object_created_notification(self):
         n.delete()
 
 
+class TestTransactionNotifications(BaseTestTransactionNotifications):
+    def _create_test_app(self):
+        org = self._get_org()
+        test_app = TestApp.objects.create(name='test_app', organization=org)
+        return test_app
+
+    def _get_test_app(self, test_app_name='test_app'):
+        try:
+            return TestApp.objects.get(name=test_app_name)
+        except TestApp.DoesNotExist:
+            return self._create_test_app()
+
+    def test_test_app_cache_invalidation(self):
+        test_app = self._get_test_app()
+        content_type = ContentType.objects.get_for_model(test_app._meta.model)
+        cache_key = Notification._cache_key(content_type.id, test_app.id)
+        test_app_cache = cache.get(cache_key, None)
+        self.assertEqual(test_app_cache.name, test_app.name)
+        test_app.name = 'new test app'
+        test_app.save()
+        notification = Notification.objects.get(target_content_type=content_type)
+        self.assertEqual(notification.target.name, test_app.name)
```

Comment:
```suggestion
        self.assertEqual(notification.target.name, test_app.name)
        test_app_cache = cache.get(cache_key, None)
        self.assertEqual(test_app_cache.name, test_app.name)
```

You need to test the value in cache as well.